lib/sax.js

```diff
@@ -120,12 +120,11 @@ function parse(source,defaultNSMapCopy,entityMap,domBuilder,errorHandler){
 				var tagName = source.substring(tagStart + 2, end).replace(/[ \t\n\r]+$/g, '');
 				var config = parseStack.pop();
 				if(end<0){
-
-	        		tagName = source.substring(tagStart+2).replace(/[\s<].*/,'');
+	        		tagName = source.substring(tagStart+2).replace(/[\s<].*/g,'');
 	        		errorHandler.error("end tag name: "+tagName+' is not complete:'+config.tagName);
 	        		end = tagStart+1+tagName.length;
 	        	}else if(tagName.match(/\s</)){
-	        		tagName = tagName.replace(/[\s<].*/,'');
+	        		tagName = tagName.replace(/[\s<].*/g,'');
```

Comment:
By calculating the length first, the injection could be prevented.
Snapshot test 'unclosed inner' in simple.test.js will start failing, but while looking into it I'm struggling to understand why the test should work without errors. My expectation would be that it fails with an error.

```suggestion
	        		end = tagStart + 2 + tagName.length;
	        		tagName = tagName.replace(/[\s<].*/g,'');
```

Reply:
And we need to shift the index +1 to catch the closing '>'